ansible_ai_connect_chatbot/src/useChatbot/useChatbot.ts

```diff
@@ -146,25 +146,45 @@ export const useChatbot = () => {
   >(undefined);
   const [selectedModel, setSelectedModel] = useState("granite3-8b");
 
-  const addMessage = (newMessage: ExtendedMessage) => {
-    setMessages((msgs: ExtendedMessage[]) => [...msgs, newMessage]);
+  const addMessage = (
+    newMessage: ExtendedMessage,
+    addAfter?: ExtendedMessage,
+  ) => {
+    setMessages((msgs: ExtendedMessage[]) => {
+      const newMsgs: ExtendedMessage[] = [];
+      newMessage.scrollToHere = true;
+      let inserted = false;
+      for (const msg of msgs) {
+        delete msg.scrollToHere;
```

Comment:
It might be _nicer_ to set to `false` rather than delete from the object?

Reply:
I thought having one `scrollToHere` flag in the whole messages would be simpler, but it is true that changing a boolean value should be less expensive then deleting the attribute... I will update the code as you suggested.

Reply:
Updated the code in the latest commit. Thanks!